packages/docusaurus-plugin-openapi/src/sidebars/index.ts

```diff
@@ -5,15 +5,74 @@
  * LICENSE file in the root directory of this source tree.
  * ========================================================================== */
 
-import { ApiMetadata, ApiPageMetadata } from "../types";
+import path from "path";
+
+import { NavbarItem } from "@docusaurus/theme-common";
+import _ from "lodash";
+
+import { ApiPageMetadata } from "../types";
 
 interface Options {
   sidebarCollapsible: boolean;
   sidebarCollapsed: boolean;
 }
 
+export type BaseItem = {
+  [key: string]: any;
+  title: string;
+  permalink: string;
+  id: string;
+  source: string;
+};
+
+export type InfoItem = BaseItem & {
+  type: "info";
+  info: any;
+};
+
+export type ApiItem = BaseItem & {
+  type: "api";
+  api: {
+    info?: {
+      title?: string;
+    };
+    tags?: string[] | undefined;
+  };
+};
+
+type Item = InfoItem | ApiItem;
+
+export function generateSidebars(
+  items: Item[],
+  options: Options
+): NavbarItem[] {
+  const sections = _(items)
+    .groupBy((item) => item.source)
+    .mapValues((items, source) => {
+      const prototype = items.find((x) => {
+        return x.api?.info != null;
+      });
+      const info = prototype?.api?.info;
+      const fileName = path.basename(source).split(".")[0];
+      return {
+        ...options,
+        type: "category",
+        label: info?.title || fileName,
+        items: groupByTags(items, options),
+      };
+    })
+    .values()
+    .value();
+
+  if (sections.length === 1) {
+    return sections[0].items;
+  }
```

Comment:
Ohhh I see now, this would circumvent the single spec issue I was talking about on our call